distribution/lib/Standard/Microsoft/0.0.0-dev/src/Internal/SQLServer_Dialect.enso

```diff
@@ -302,13 +309,69 @@ type SQLServer_Dialect
         if raw_sql.contains "#" . not then
             jdbc.ensure_query_has_no_holes raw_sql
 
+    ## PRIVATE
+       Returns a pair of a SQL_Builder for the given expression and a vector of columns
+       that have been used in the expression and need to be checked for nulls.
+    generate_expression self base_gen expr = case expr of
```

Comment:
I don't love that some of the `Base_Generator` logic is spreading into the dialect here. Would it be possible for `Base_Generator.generate_expression` to have extra logic to handle this special case? Most of the clauses here return the main result value as the second return value.

Reply:
No I don't like it either. But I don't like the idea of adding this complexity into the base either.

Reply:
I agree with Ad - the Base should be structured in a way to be extensible, but dialect-specific logic shouldn't leak there (too much), but be placed in each dialect. So I think this is OK to keep here.

Reply:
I think this complex handling of NULLs deserves some explanation in the dialect documentation: why we need it and how we are doing it.

Reply:
Added a more detailed comment